src/circuit_writer/ir.rs

```diff
@@ -501,6 +501,34 @@ impl<B: Backend> IRWriter<B> {
                 return Ok(Some(var));
             }
             StmtKind::Comment(_) => (),
+            StmtKind::Ite {
+                condition,
+                then_branch,
+                else_branch,
+            } => {
+                fn_env.nest();
+                let cond_var = self
+                    .compute_expr(fn_env, condition)?
+                    .unwrap()
+                    .value(self, fn_env)
+                    .cvars[0]
+                    .clone();
+                let ret = self.compile_block(fn_env, then_branch)?.unwrap().cvars[0].clone();
```

Comment:
why then_branch block isn't handled the same way as the else_branch block?

Reply:
shouldn't the fn_env.nest() and fn_env.pop() be called for each `compile_block`?